### DISCUSSION TITLE
Img2img batch generation

### DISCUSSION THREAD

**Discussion by popsoda27** [vladmandic/sdnext#2042] [General]
Not sure if this is a bug or some setting that I'd overlooked, but whenever I do batch img2img, it keeps churning duplicate images for each image I have in the source folder, ending up having 2 of each.

Is there something that I missed? I have gone through the settings over and over and I still can't figure out what is causing the duplicates. I made sure that the batch size and count are both at 1.

**Comment by josh5671** [comment]
Try search 'before' in the setting.
there should be 4 setting related 
```
Save copy of image before applying highres fix
Save copy of image before running refiner
Save copy of image before doing face restoration
Save copy of image before applying color correction
```

suspect one of these causing your 'problem'

**Comment by popsoda27** [comment]
> Try search 'before' in the setting. there should be 4 setting related
> 
> ```
> Save copy of image before applying highres fix
> Save copy of image before running refiner
> Save copy of image before doing face restoration
> Save copy of image before applying color correction
> ```
> 
> suspect one of these causing your 'problem'

Thank you for your reply. I confirm that none of the above options are enabled.
![image](https://github.com/vladmandic/automatic/assets/20932783/00e2ecf0-13be-43e7-99dc-6890042e3a7f)

**Comment by vladmandic** [comment]
I've tried to reproduce the problem, no luck.
Can you show some generated file names, perhaps that will point in a specific direction.

**Reply by popsoda27** [reply]
![image](https://github.com/vladmandic/automatic/assets/20932783/11df8c86-70cf-4554-9dd4-b2d6570236fa)
Does any of these enabled options give an indication that it might save duplicate images?

**Comment by popsoda27** [comment]
Tried sending a single txt2img to img2img...generates a single img2img webp.
Tried batch img2img sourcing from txt2img folder containing 4 images....generates double img2img for each txt2img.

The filenames goes like this...
00000-photo of a woman wearing blue dress squniform.webp
00001-.webp
00002-photo of a woman wearing blue dress squniform.webp
00003-.webp
00004-photo of a woman wearing blue dress squniform.webp
00005-.webp
00006-photo of a woman wearing blue dress squniform.webp
00007-.webp

00000 & 00001 are the same, 00002 & 00003 the same and so on.

Maybe file format? So I changed to png and restarted the batch. Same thing duplicate images

**Comment by vladmandic** [comment]
So duplicate is saved AFTER actual generate, not before. And at that point core SDnext is pretty much done.
What are the extension you have installed? Perhaps one of them does the duplicate save?

**Reply by popsoda27** [reply]
other than the built-in ones, here are the extensions I had installed.
![image](https://github.com/vladmandic/automatic/assets/20932783/37fa75f7-ac3a-4eec-8dab-b77ed7682372)

**Reply by popsoda27** [reply]
the only extension that has anything to do with saving images would be ADetailer.
![image](https://github.com/vladmandic/automatic/assets/20932783/1526515b-1795-4c7c-9e11-0b6b2f6dc2be)

but I only use it in txt2img and not img2img.

**Comment by popsoda27** [comment]
After much tinkering, I like to confirm that by NOT setting any default upscaler under the Upscaling menu options resolved the issue for me.
![image](https://github.com/vladmandic/automatic/assets/20932783/868148dc-053a-4061-a03b-a17cd0da4050)

This can be closed.
